platforms/ide/problems-api/src/main/java/org/gradle/internal/problems/DefaultProblemDiagnosticsFactory.java

```diff
@@ -105,7 +109,8 @@ private ProblemDiagnostics locationFromStackTrace(@Nullable Throwable throwable,
         Location location = null;
         if (throwable != null) {
             stackTrace = transformer.transform(throwable.getStackTrace());
-            location = locationAnalyzer.locationForUsage(stackTrace, fromException);
+            Failure stackTracingFailure = failureFactory.create(throwable);
+            location = locationAnalyzer.locationForUsage(stackTracingFailure, fromException);
```

Comment:
💭  This is the main change, temporarily creating a failure to get a classified stacktrace and use it to extract the location.

The follow-up changes would use the created `Failure` and expose it in the `ProblemDiagnostics`